rust/src/wallet.rs

```diff
@@ -520,19 +521,20 @@ pub async fn open_channel(peer_info: PeerInfo, taker_amount: u64) -> Result<()>
 /// If the first channel is not usable, it might be because we've lost
 /// the connection with the 10101 maker, according to the
 /// `rust-lightning` logs.
-pub fn is_first_channel_usable() -> Result<bool> {
-    let channel_manager = {
-        let lightning = &get_wallet()?.lightning;
-
-        lightning.channel_manager.clone()
-    };
-
-    let is_usable = match channel_manager.list_channels().first() {
+pub fn is_first_channel_usable() -> bool {
+    match get_first_channel_details() {
         Some(channel_details) => channel_details.is_usable,
-        None => return Ok(false),
-    };
+        None => false,
+    }
+}
+
+pub fn get_first_channel_details() -> Option<ChannelDetails> {
+    let channel_manager = match &get_wallet() {
+        Ok(wallet) => Some(wallet.lightning.channel_manager.clone()),
+        Err(_) => None,
+    }?;
```

Comment:
nit: see below (gh didn't allow me doing it in one because the empty line on line 536 was there before 😂)
```suggestion

```